flipperdevices/flipperzero-firmware
# [FL-3456] Allow for larger Infrared remotes

# What's new

- Do not load all infrared signals into RAM at once, load them on-demand instead.
- Improve general error handling and UI/UX
- Add doxygen comments
- Basically nearly full refactoring

# Verification 

1. Download [Daikin FTXC35DV1B Full.ir.tar.gz](https://github.com/flipperdevices/flipperzero-firmware/files/13113501/Daikin.FTXC35DV1B.Full.ir.tar.gz), unzip it and put into the `/ext/infrared` directory on the Flipper.
2. Load it using `Infrared -> Saved Remotes
3. The application should not crash
4. Test adding, renaming, moving and deleting signals since this functionality was reimplemented from scratch.

# Checklist (For Reviewer)
